### PR TITLE
[CAI-41] Chat component

### PR DESCRIPTION
**Pull request by Sebastiano-Bertolin**
Pending on [PR-952](https://github.com/pagopa/developer-portal/pull/952)
and on [PR-951](https://github.com/pagopa/developer-portal/pull/951)

#### List of Changes
Add Chat component 

#### Motivation and Context
This component is used in the Chatbot modal and contains the messages and input text field

#### How Has This Been Tested?
Tested via Storybook

#### Screenshots (if appropriate):
Screenshot 2024-07-03 at 16 44 06

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/components/molecules/Chat/Chat.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+            transformOrigin={{
+              vertical: 'top',
+              horizontal: 'right',
+            }}
+          >
+            <MenuItem onClick={handleClose}>
+              <ListItemIcon>
+                <Delete fontSize='small' />
+              </ListItemIcon>
+              <ListItemText>Delete</ListItemText>
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/molecules/Chat/Chat.tsx:82]: This should be translated

File: apps/nextjs-website/src/components/molecules/Chat/Chat.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            </MenuItem>
+            <MenuItem onClick={handleClose}>
+              <ListItemIcon>
+                <History fontSize='small' />
+              </ListItemIcon>
+              <ListItemText>Cronologia Chat</ListItemText>
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/molecules/Chat/Chat.tsx:88]: This should be translated

File: apps/nextjs-website/src/components/molecules/Chat/Chat.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  const open = Boolean(anchorEl);
+
+  const scrollRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (scrollRef.current) {
+      scrollRef.current.scrollIntoView({ behavior: 'smooth' });
+    }
+  }, [chatMessages]);
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/components/molecules/Chat/Chat.tsx:33]: `handleClick` name isn't very expressive maybe something like `handleChatMenuClick` could be more clear ?

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: .changeset/popular-glasses-love.md
Comment (MarcoPonchia) [inline .changeset/popular-glasses-love.md:1]: This changeset can be deleted, is already present in this PR https://github.com/pagopa/developer-portal/pull/952/files

Comment (MarcoPonchia) [reply .changeset/popular-glasses-love.md:1]: Maybe just align branch to main will remove this changeset

Comment (Sebastiano-Bertolin) [reply .changeset/popular-glasses-love.md:1]: Done

Comment (MarcoPonchia) [inline .changeset/popular-glasses-love.md:1]: I think this file should be deleted

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Pull request merged**